### DISCUSSION TITLE
Client Side Rendering Not Working In Dev Mode

### DISCUSSION THREAD

**Discussion by ahurlburt** [vercel/next.js#15782] [Help] [answered]
# Bug report

## Describe the bug

I'm using a custom server implementation of next. I am having an issue in dev mode which is not reproducible outside of dev mode.

Basically, when I launch next in dev mode then the client React methods are not called (ie ComponentDidMount)

const app = next({ dev, quiet: false });

When I launch the app without dev mode after running 'next build':

const app = next({ quiet: false });

Then the client side lifecycle methods are called appropriately.

I'm guessing I'm probably doing something wrong somewhere in my application but I am new to next and any help / pointing to the right direction would be appreciated. Please let me know what other information may be relevant to help reproduce.

I recently upgraded next, it was previously working (on 8.1.0).

## Expected behavior

I would expect dev mode to call the same lifecycle as the production build.

## System information

- OS: macOS
- Version of Next.js: [9.5.0]
- Version of Node.js: [10.16.0]

**Comment by jamesmosier** [comment]
Hi @ahurlburt. Do you have an example repository of the code that is not working?

**Comment by ahurlburt** [comment, marked as answer]
Thanks @jamesmosier for your response. I didn't include a link to my repo as it is a private app for a bustiness. It is also a shopify app so I wasn't sure how easily it would be to reproduce without shopify config (which requires client id/secret).

That being said.. after a fresh start of everything and cleaning and reinstalling dependencies this seems to be now working as expected. So I guess I'll close this for now and if it recurs try to provide a reproducible example.
